Regression in nightly handling features + debug assertions

**Problem**

`1.43.0-nightly (4ad624882 2020-03-03)` has regressed handling of debug assertions with features.

**Steps**
1. Install `1.43.0-nightly (4ad624882 2020-03-03)`

2. Build a crate with the following Cargo.toml

```
[package]
name = "test-bin"
version = "0.1.0"
edition = "2018"

[dependencies]
rocket_contrib = { version = "0.4.2", features = [ "templates", "tera_templates" ] }
```

3. Build fails with:

```
   Compiling rocket_contrib v0.4.3
error[E0463]: can't find crate for `notify`
  --> /opt/rust/registry/src/crates-io-27bb2a0041c1489d/rocket_contrib-0.4.3/src/templates/fairing.rs:35:5
   |
35 |     extern crate notify;
   |     ^^^^^^^^^^^^^^^^^^^^ can't find crate

error: aborting due to previous error

For more information about this error, try `rustc --explain E0463`.
error: could not compile `rocket_contrib`.

To learn more, run the command again with --verbose.
```

Cargo has not downloaded the required crate, so I assume a Cargo bug, instead of a rustc bug.

4. Build works with `cargo 1.43.0-nightly (e57bd0299 2020-02-21)`

**Possible Solution(s)**

Dependency is pulled in via a debug assertion:

https://github.com/SergioBenitez/Rocket/blob/v0.4.2/contrib/lib/Cargo.toml#L79-L80

`notify` is not downloaded by Cargo during a clean build.

`Cargo.lock` does reference `notify`:

```
[[package]]
name = "rocket_contrib"
version = "0.4.3"
source = "registry+https://github.com/rust-lang/crates.io-index"
checksum = "66926894ef94396204c53885f54a1b0e94a88ce6c9478c2fd403261fc0efb523"
dependencies = [
 "glob 0.3.0",
 "log 0.4.8",
 "notify",
 "rocket",
 "serde",
 "serde_json",
 "tera",
]
```

This is a recent regression - `rust version 1.43.0-nightly (d3c79346a 2020-02-29)` also works.

Suspect https://github.com/rust-lang/cargo/pull/7962 is at fault,.

--- a/src/cargo/core/compiler/build_context/target_info.rs
+++ b/src/cargo/core/compiler/build_context/target_info.rs
@@ -192,7 +192,10 @@ impl TargetInfo {
 
     fn not_user_specific_cfg(cfg: &CargoResult<Cfg>) -> bool {
         if let Ok(Cfg::Name(cfg_name)) = cfg {
-            if cfg_name == "debug_assertions" || cfg_name == "proc_macro" {
+            // This should also include "debug_assertions", but it causes
+            // regressions. Maybe some day in the distant future it can be
+            // added (and possibly change the warning to an error).
+            if cfg_name == "proc_macro" {
                 return false;
             }
         }

--- a/src/cargo/core/compiler/custom_build.rs
+++ b/src/cargo/core/compiler/custom_build.rs
@@ -228,6 +228,11 @@ fn build_work<'a, 'cfg>(cx: &mut Context<'a, 'cfg>, unit: &Unit<'a>) -> CargoRes
         }
     }
     for (k, v) in cfg_map {
+        if k == "debug_assertions" {
+            // This cfg is always true and misleading, so avoid setting it.
+            // That is because Cargo queries rustc without any profile settings.
+            continue;
+        }
         let k = format!("CARGO_CFG_{}", super::envify(&k));
         match v {
             Some(list) => {

